crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1527,7 +1527,7 @@ macro_rules! impl_anytuple_fetch {
                     if _not_first {
```

Comment:
Github doesn't let me select the lines above, but there's a comment claiming that the `with`/`without` access should be added only if all the params have that access, but this is no longer true, instead disjoint access is properly tracked with `append_or`.

Reply:
Removed the comment and renamed variables a bit to avoid confusion with the "intersect" word